tests/trainer/test_trainer.py

```diff
@@ -2409,6 +2410,128 @@ def test_end_to_end_example(self):
             execute_subprocess_async(command)
             # successful return here == success - any errors would have caused an error or a timeout in the sub-call
 
+    def test_accelerator_config_from_dict(self):
```

Comment:
What's the expected behaviour when no accelerator config is passed in? This should be tested